funcx_endpoint/funcx_endpoint/endpoint/interchange.py

```diff
@@ -257,13 +269,24 @@ def start(self):
             else:
                 log.debug("Starting threads and main loop")
 
+            if self.time_to_quit:
```

Comment:
You just forgot to check this flag previously, right?  The previous check only in process_pending_tasks.

Because it is related to kill_event.set(), and the logic path is similar, (and they should be handled similarly), is it simpler to use the same mechanism instead of two similar variables?  But that's separate to this PR, though.

Reply:
> You just forgot to check this flag previously, right?

That's exactly correct.

> But that's separate to this PR, though.

Your analysis is solid, but I arrived here via an ancillary path.  I would like to address exactly that, but not in this PR.  Tangential change, that.